Command/ProcessEventCommand.php

```diff
@@ -53,34 +67,45 @@ protected function configure()
     {
         $this
             ->setName('itkg_delay_event:process')
-            ->setDescription('Process async events')
-            ->addArgument('n', InputArgument::OPTIONAL, 'Number of events to process', 1);
+            ->setDescription('Process async events');
     }
 
     /**
      * @param InputInterface  $input
      * @param OutputInterface $output
      *
+     * @throws LockException
+     * @throws \Exception
+     *
      * @return int|null|void
      */
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        $events = $this->eventRepository->findBy(array(), array('createdAt' => 1), $input->getArgument('n'));
+        if ($this->lockHandler->isLocked()) {
+            throw new LockException('Command is locked by another processus');
+        }
 
-        /** @var Event $event */
-        foreach ($events as $event) {
-            try {
-                /** @TODO : ADD LOGS */
-                $event->setDelayed(false);
+        $this->lockHandler->lock();
 
+        $exception = null;
+        try {
+            while (true) {
+                $event = $this->eventRepository->findFirstTodoEvent();
+                if (!$event) {
+                    break;
+                }
+                $event->setDelayed(false);
                 $this->eventProcessor->process($event);
                 $this->eventManager->delete($event);
-            } catch(\Exception $e) {
-                // @TODO : Manage rollback count maybe
-                $event->rollback();
-                die($e->getMessage());
-                $this->eventManager->save($event);
+
             }
+        } catch(\Exception $e) {
+            $exception = $e;
+        }
+        $this->lockHandler->release();
+
+        if ($exception) {
+            throw $exception;
```

Comment:
I think we need to `$this->lockHandler->release();` but we don't have "finally" in php 5.4...